src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -113,125 +137,269 @@ internal abstract class AbstractAddParameterCodeFixProvider<
+            if (method.ExplicitInterfaceImplementations.Length > 0)
+            {
+                return true;
             }
+
+            // For implicit interface implementations lets check if the characteristic of the method
+            // allows it to implicit implement an interface member.
+            if (method.DeclaredAccessibility == Accessibility.Private || method.DeclaredAccessibility == Accessibility.NotApplicable)
+            {
+                return false;
+            }
+
+            if (method.IsStatic)
+            {
+                return false;
+            }
+
+            // Now check if the method does implement an interface member
+            if (method.ExplicitOrImplicitInterfaceImplementations().Length > 0)
+            {
+                return true;
+            }
+
+            return false;
         }
 
-        private int NonParamsParameterCount(IMethodSymbol method)
-            => method.IsParams() ? method.Parameters.Length - 1 : method.Parameters.Length;
+        private static string GetCodeFixTitle(string resourceString, IMethodSymbol methodToUpdate, IEnumerable<string> parameters)
+        {
+            var methodPrefix = methodToUpdate.IsConstructor()
+                ? ""
+                : $"{methodToUpdate.ContainingType.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat)}.";
+            var signature = $"{methodPrefix}{methodToUpdate.Name}({string.Join(", ", parameters)})";
+            var title = string.Format(resourceString, signature);
+            return title;
+        }
 
-        private async Task<Document> FixAsync(
+        private async Task<Solution> FixAsync(
             Document invocationDocument,
             IMethodSymbol method,
             TArgumentSyntax argument,
             SeparatedSyntaxList<TArgumentSyntax> argumentList,
+            bool fixAllReferences,
             CancellationToken cancellationToken)
         {
-            var methodDeclaration = await method.DeclaringSyntaxReferences[0].GetSyntaxAsync(cancellationToken).ConfigureAwait(false);
+            var solution = invocationDocument.Project.Solution;
+            var argumentType = await GetArgumentTypeAsync(invocationDocument, argument, cancellationToken).ConfigureAwait(false);
+            // the argumentNameSuggestion is the base for the parameter name. For each method declaration the name is made unique to avoid name collisions.
+            var (argumentNameSuggestion, isNamedArgument) = await GetNameSuggestionForArgumentAsync(invocationDocument, argument, cancellationToken).ConfigureAwait(false);
+            var referencedSymbols = fixAllReferences
+                ? await FindMethodDeclarationReferences(invocationDocument, method, cancellationToken).ConfigureAwait(false)
+                : method.GetAllMethodSymbolsOfPartialParts();
+            // TODO: Insert hint in the fix with a warning if anySymbolsReferenceNotInSource is true
```

Comment:
nit: consider adding newlines before these comments to help break things up and make things easier to read.